crates/ark/src/interface.rs

```diff
@@ -536,7 +574,12 @@ impl RMain {
 
             log::info!("Sending kernel info: {version}");
             self.kernel_init_tx.broadcast(kernel_info);
+
+            // Internal initialisation flag, should only be used on the R thread
             self.initializing = false;
+
+            // Used as thread-safe initialisation flag
+            R_INIT.set(()).unwrap();
```

Comment:
That would make sense but we'd get into naming conflicts with the static methods so probably not worth changing.

This is only used in `WriteConsole` so I've decided we don't need to worry about synchronisation overhead here, and we now use the thread-safe `RMain::is_r_initialized()` instead of the boolean flag to make things simpler.